### PR TITLE
feat: add compoundAllForUsers

### PR DESCRIPTION
**Pull request by Cphilo**

### CODE REVIEW

File: contracts/WooStakingManager.sol [hunk 1]
```diff
@@ -203,6 +203,40 @@ contract WooStakingManager is IWooStakingManager, BaseAdminOperation, Reentrancy
...
+            uint256 wooTotalAmount = 0;
+            address rewardToken = _rewarder.rewardToken();
+            if (rewardToken == woo) {
+                wooTotalAmount = totalReward;
+            } else {
+                wooTotalAmount = wooPP.swap(rewardToken, woo, totalReward, 0, selfAddr, selfAddr);
+            }
+            TransferHelper.safeApprove(woo, address(stakingLocal), wooTotalAmount);
+            for (uint256 j = 0; j < len; ++j) {
+                stakingLocal.stake(_users[j], (wooTotalAmount * rewards[j]) / totalReward);
```
Comment (merlincdj) [inline contracts/WooStakingManager.sol:235]: if `rewarders.length()` >= 2, then `stakingLocal.stake()` will be called `rewarders.length() * len` times, causing the huge gas cost. can we just simplify `len` times

Comment (fb-alexcq) [reply contracts/WooStakingManager.sol:235]: Probably better to store in a local memory array?

Comment (Cphilo) [reply contracts/WooStakingManager.sol]: Resolved

File: contracts/WooStakingManager.sol [hunk 2]
```diff
@@ -203,6 +203,40 @@ contract WooStakingManager is IWooStakingManager, BaseAdminOperation, Reentrancy
...
+        uint256 len = _users.length;
+        unchecked {
+            for (uint256 i = 0; i < len; ++i) {
+                compoundMP(_users[i]);
+            }
+        }
+
+        address selfAddr = address(this);
+        uint256[] memory rewards = new uint256[](len);
+        for (uint256 i = 0; i < rewarders.length(); ++i) {
```
[the remark this answers is not in the document]
Comment (Cphilo) [reply contracts/WooStakingManager.sol]: Resolved

**Review by merlincdj — APPROVED** [review]

File: contracts/WooStakingLocal.sol [hunk 3]
```diff
@@ -70,6 +70,21 @@ contract WooStakingLocal is IWooStakingLocal, BaseAdminOperation, ReentrancyGuar
         _stake(_user, _amount);
     }
 
+    function stakeForUsers(address[] memory _users, uint256[] memory _amounts) external whenNotPaused onlyAdmin {
```
Comment (fb-alexcq) [inline contracts/WooStakingLocal.sol:73]: maybe add a totalWooAmount as the parameter?

**Review by fb-alexcq — APPROVED** [review]

File: contracts/WooStakingLocal.sol (hunk 3 above)
Comment (Cphilo) [reply contracts/WooStakingLocal.sol]: Resolved

**Pull request merged**